lib/api/controllers/writeController.js

```diff
@@ -13,295 +18,219 @@ function WriteController (kuzzle) {
   /**
    * Create a new document
    *
-   * @param {RequestObject} requestObject
-   * @param {Object} userContext
-   * @returns {Promise}
+   * @param {Request} request
+   * @returns {Promise<Object>}
    */
-  this.create = function writeCreate (requestObject, userContext) {
-    var
-      modifiedData = null;
+  this.create = function writeCreate (request) {
+    /** @type KuzzleRequest */
+    var modifiedRequest;
 
-    requestObject.userId = userContext.token.userId;
+    assertBody(request, 'create');
+    assertIndexAndCollection(request, 'create');
 
-    return requestObject.isValid()
-      .then(() => {
-        return kuzzle.pluginsManager.trigger('data:beforeCreate', {requestObject, userContext});
-      })
-      .then(data => {
-        modifiedData = data;
+    return kuzzle.validation.validationPromise(request, false)
+      .then(newRequest => {
+        modifiedRequest = newRequest;
 
-        return kuzzle.validation.validate(modifiedData.requestObject);
-      })
-      .then(modifiedRequestObject => {
-        modifiedData.requestObject = modifiedRequestObject;
+        request.input.args.state = 'pending';
+        kuzzle.notifier.publish(modifiedRequest);
 
-        return engine.create(modifiedData.requestObject);
+        return engine.create(modifiedRequest);
       })
       .then(response => {
-        kuzzle.notifier.notifyDocumentCreate(modifiedData.requestObject, response);
+        kuzzle.notifier.notifyDocumentCreate(modifiedRequest, response);
 
-        return kuzzle.pluginsManager.trigger('data:afterCreate', {
-          responseObject: new ResponseObject(modifiedData.requestObject, response),
-          userContext: modifiedData.userContext
-        });
+        return Promise.resolve(response);
```

Comment:
This kind of return statement is quite systematic in Kuzzle but I don't think it is useful. AFAIK, within the **last** `then` callback, returning a value is equivalent to returning a Promise resolving a value. Please, tell me if I'm wrong, cuz I need to know if I'm totally next to the plaque...